### ISSUE TITLE
Bug: Uninitialized value on degraded raid with arcconf

### ISSUE THREAD

**Issue by wienczny** [glensc/nagios-plugin-check_raid#155]
**Version**
```check_raid Version 3.2.1```

**Output of `check_raid -d`:**

```
PATH=/usr/StorMan /usr/lib/nagios/plugins/check_raid  -d                                                                                                                                        
DEBUG EXEC: /proc/mdstat at /usr/lib/nagios/plugins/check_raid line 435.                                                                                                                                          
DEBUG EXEC: /usr/StorMan/arcconf GETSTATUS 1 at /usr/lib/nagios/plugins/check_raid line 435.
DEBUG EXEC: /usr/StorMan/arcconf GETCONFIG 1 AL at /usr/lib/nagios/plugins/check_raid line 435.
Unparsed Physical Device data: [         Model                              : ] at /usr/lib/nagios/plugins/check_raid line 2661, <$fh> line 128.
Use of uninitialized value $s[0] in join or string at /usr/lib/nagios/plugins/check_raid line 314.
CRITICAL: arcconf:[Controller:Optimal, Defunct drives:1, Logical Device 0(HDD):Optimal, Logical Device 1(SSD):Degraded, Drives: =Failed WD-WMC4M1057040,WD-WMC4M1019533,50026B726900BD5E,WD-WMC4M1311378,WD-WMC4M1081289,50026B726900BD87,50026B726900BD5F=Online]
```

**Output of `/usr/StorMan/arcconf GETSTATUS 1`:**
```
Controllers found: 1
   Current operation              : None

Command completed successfully.
```

**Output of `/usr/StorMan/arcconf GETCONFIG 1 AL`:**
```
Controllers found: 1
----------------------------------------------------------------------
Controller information
----------------------------------------------------------------------
   Controller Status                        : Optimal
   Channel description                      : SAS/SATA
   Controller Model                         : Adaptec 6805E
   Controller Serial Number                 : 3A211320B60
   Physical Slot                            : 17
   Temperature                              : 61 C/ 141 F (Normal)
   Installed memory                         : 128 MB
   Copyback                                 : Disabled
   Background consistency check             : Disabled
   Automatic Failover                       : Enabled
   Global task priority                     : High
   Performance Mode                         : Default/Dynamic
   Stayawake period                         : Disabled
   Spinup limit internal drives             : 0
   Spinup limit external drives             : 0
   Defunct disk drive count                 : 1
   Logical devices/Failed/Degraded          : 2/0/1
   NCQ status                               : Enabled
   Statistics data collection mode          : Enabled
   --------------------------------------------------------
   Controller Version Information
   --------------------------------------------------------
   BIOS                                     : 5.2-0 (19109)
   Firmware                                 : 5.2-0 (19109)
   Driver                                   : 1.2-1 (41010)
   Boot Flash                               : 5.2-0 (19109)

----------------------------------------------------------------------
Logical device information
----------------------------------------------------------------------
Logical device number 0
   Logical device name                      : HDD
   RAID level                               : 10
   Status of logical device                 : Optimal
   Size                                     : 3809270 MB
   Stripe-unit size                         : 256 KB
   Read-cache mode                          : Enabled
   Write-cache mode                         : Enabled (write-back)
   Write-cache setting                      : Enabled (write-back)
   Partitioned                              : Yes
   Protected by Hot-Spare                   : No
   Bootable                                 : Yes
   Failed stripes                           : No
   Power settings                           : Disabled
   --------------------------------------------------------
   Logical device segment information
   --------------------------------------------------------
   Group 0, Segment 0                       : Present (0,0)      WD-WMC4M1057040
   Group 0, Segment 1                       : Present (0,1)      WD-WMC4M1019533
   Group 1, Segment 0                       : Present (0,4)      WD-WMC4M1311378
   Group 1, Segment 1                       : Present (0,5)      WD-WMC4M1081289

Logical device number 1
   Logical device name                      : SSD
   RAID level                               : 10
   Status of logical device                 : Degraded
   Size                                     : 974838 MB
   Stripe-unit size                         : 256 KB
   Read-cache mode                          : Enabled
   Write-cache mode                         : Enabled (write-back)
   Write-cache setting                      : Enabled (write-back)
   Partitioned                              : Yes
   Protected by Hot-Spare                   : No
   Bootable                                 : No
   Failed stripes                           : No
   Power settings                           : Disabled
   --------------------------------------------------------
   Logical device segment information
   --------------------------------------------------------
   Group 0, Segment 0                       : Inconsistent (0,2) 
   Group 0, Segment 1                       : Present (0,3) 50026B726900BD5E
   Group 1, Segment 0                       : Present (0,6) 50026B726900BD87
   Group 1, Segment 1                       : Present (0,7) 50026B726900BD5F


----------------------------------------------------------------------
Physical Device information
----------------------------------------------------------------------
      Device #0
         Device is a Hard drive
         State                              : Online
         Supported                          : Yes
         Transfer Speed                     : SATA 6.0 Gb/s
         Reported Channel,Device(T:L)       : 0,0(0:0)
         Reported Location                  : Connector 0, Device 0
         Vendor                             : WDC
         Model                              : WD20EFRX-68EUZN0
         Firmware                           : 80.00A80
         Serial number                      : WD-WMC4M1057040
         Size                               : 1907729 MB
         Write Cache                        : Enabled (write-back)
         FRU                                : None
         S.M.A.R.T.                         : No
         S.M.A.R.T. warnings                : 0
         Power State                        : Full rpm
         Supported Power States             : Full rpm,Powered off
         NCQ status                         : Disabled
      Device #1
         Device is a Hard drive
         State                              : Online
         Supported                          : Yes
         Transfer Speed                     : SATA 6.0 Gb/s
         Reported Channel,Device(T:L)       : 0,1(1:0)
         Reported Location                  : Connector 0, Device 1
         Vendor                             : WDC
         Model                              : WD20EFRX-68EUZN0
         Firmware                           : 80.00A80
         Serial number                      : WD-WMC4M1019533
         Size                               : 1907729 MB
         Write Cache                        : Enabled (write-back)
         FRU                                : None
         S.M.A.R.T.                         : No
         S.M.A.R.T. warnings                : 0
         Power State                        : Full rpm
         Supported Power States             : Full rpm,Powered off
         NCQ status                         : Disabled
      Device #2
         Device is a Hard drive
         State                              : Failed
         Supported                          : Yes
         Reported Channel,Device(T:L)       : 0,2(2:0)
         Reported Location                  : Connector 0, Device 2
         Vendor                             : *MISSING*
         Model                              : 
         Firmware                           : 
         Size                               : 0 MB
         Write Cache                        : Unknown
         FRU                                : None
         S.M.A.R.T.                         : No
         S.M.A.R.T. warnings                : 0
      Device #3
         Device is a Hard drive
         State                              : Online
         Supported                          : Yes
         Transfer Speed                     : SATA 6.0 Gb/s
         Reported Channel,Device(T:L)       : 0,3(3:0)
         Reported Location                  : Connector 0, Device 3
         Vendor                             : KINGSTON
         Model                              :  SKC400S37512G
         Firmware                           : SAFM00.Y
         Serial number                      : 50026B726900BD5E
         Size                               : 488386 MB
         Write Cache                        : Enabled (write-back)
         FRU                                : None
         S.M.A.R.T.                         : No
         S.M.A.R.T. warnings                : 0
         Power State                        : Full rpm
         Supported Power States             : Full rpm,Powered off,Reduced rpm
         NCQ status                         : Disabled
      Device #4
         Device is a Hard drive
         State                              : Online
         Supported                          : Yes
         Transfer Speed                     : SATA 6.0 Gb/s
         Reported Channel,Device(T:L)       : 0,4(4:0)
         Reported Location                  : Connector 1, Device 0
         Vendor                             : WDC
         Model                              : WD20EFRX-68EUZN0
         Firmware                           : 80.00A80
         Serial number                      : WD-WMC4M1311378
         Size                               : 1907729 MB
         Write Cache                        : Enabled (write-back)
         FRU                                : None
         S.M.A.R.T.                         : No
         S.M.A.R.T. warnings                : 0
         Power State                        : Full rpm
         Supported Power States             : Full rpm,Powered off
         NCQ status                         : Disabled
      Device #5
         Device is a Hard drive
         State                              : Online
         Supported                          : Yes
         Transfer Speed                     : SATA 6.0 Gb/s
         Reported Channel,Device(T:L)       : 0,5(5:0)
         Reported Location                  : Connector 1, Device 1
         Vendor                             : WDC
         Model                              : WD20EFRX-68EUZN0
         Firmware                           : 80.00A80
         Serial number                      : WD-WMC4M1081289
         Size                               : 1907729 MB
         Write Cache                        : Enabled (write-back)
         FRU                                : None
         S.M.A.R.T.                         : No
         S.M.A.R.T. warnings                : 0
         Power State                        : Full rpm
         Supported Power States             : Full rpm,Powered off
         NCQ status                         : Disabled
      Device #6
         Device is a Hard drive
         State                              : Online
         Supported                          : Yes
         Transfer Speed                     : SATA 6.0 Gb/s
         Reported Channel,Device(T:L)       : 0,6(6:0)
         Reported Location                  : Connector 1, Device 2
         Vendor                             : KINGSTON
         Model                              :  SKC400S37512G
         Firmware                           : SAFM00.Y
         Serial number                      : 50026B726900BD87
         Size                               : 488386 MB
         Write Cache                        : Enabled (write-back)
         FRU                                : None
         S.M.A.R.T.                         : No
         S.M.A.R.T. warnings                : 0
         Power State                        : Full rpm
         Supported Power States             : Full rpm,Powered off,Reduced rpm
         NCQ status                         : Disabled
      Device #7
         Device is a Hard drive
         State                              : Online
         Supported                          : Yes
         Transfer Speed                     : SATA 6.0 Gb/s
         Reported Channel,Device(T:L)       : 0,7(7:0)
         Reported Location                  : Connector 1, Device 3
         Vendor                             : KINGSTON
         Model                              :  SKC400S37512G
         Firmware                           : SAFM00.Y
         Serial number                      : 50026B726900BD5F
         Size                               : 488386 MB
         Write Cache                        : Enabled (write-back)
         FRU                                : None
         S.M.A.R.T.                         : No
         S.M.A.R.T. warnings                : 0
         Power State                        : Full rpm
         Supported Power States             : Full rpm,Powered off,Reduced rpm
         NCQ status                         : Disabled


Command completed successfully.
```

**Comment by glensc** [comment]
based on your test data, it seems working with current master. only thing i noticed was duplicate degraded report. i've fixed that.

**Comment by wienczny** [comment]
The raid has already been repaired. I'll try again when the next drive fails. Thanks for your quick response.

**Comment by glensc** [comment]
your problem was fixed with this commit: cf1e488e770aef04d8761907bae02d06ce3e1665
